### PR TITLE
Feature/cal 030 Companies Components

### PR DESCRIPTION
**Pull request by tomascravero**

### CODE REVIEW

**Comment by Ignacio-Sormani** [comment]
Rebase and solve conflicts before your next push

File: client/.eslintcache [hunk 1]
```diff
@@ -1 +1 @@
...
...","replacedBy":"54"},{"ruleId":"55","replacedBy":"56"},{"ruleId":"57","severity":1,"message":"58","line":2,"column":19,"nodeType":"59","messageId":"60","endLine":2,"endColumn":25},{"ruleId":"57","severity":1,"message":"61","line":11,"column":9,"nodeType":"59","messageId":"60","endLine":11,"endColumn":23},{"ruleId":"57","severity":1,"message":"62","line":1,"column":17,"nodeType":"59","messageId":"60","endLine":1,"endColumn":25},{"ruleId":"57","severity":1,"message":"63","line":10,"column":8,"nodeType":"59","messageId":"60","endLine":10,"endColumn":18},{"ruleId":"57","severity":1,"message":"64","line":13,"column":8,"nodeType":"59","messageId":"60","endLine":13,"endColumn":13},{"ruleId":"57","severity":1,"message":"65","line":15,"column":8,"nodeType":"59","messageId":"60","endLine":15,"endColumn":18},"no-native-reassign",["66"],"no-negated-in-lhs",["67"],"no-unused-vars","'Toggle' is defined but never used.","Identifier","unusedVar","'initialCompany' is assigned a value but never used.","'useState' is defined but never used.","'AddBoxIcon' is defined but never used.","'Modal' is defined but never used.","'AddCompany' is defined but never used.","no-global-assign","no-unsafe-negation"]
```
Comment (Ignacio-Sormani) [inline client/.eslintcache]: I think we shouldn't upload this file

File: client/src/components/layout/sidenav/Sidenav.jsx [hunk 2]
```diff
@@ -17,8 +17,8 @@ const Sidenav = () => {
   setInterval(UpdateTime, 1000);
 
   return (
-    <SideNav style={{ top: 80, justifyContent: 'center' }}>
-      <SideNav.Toggle style={{ marginBottom: 30 }} />
+    <SideNav style={{ top: 80, justifyContent: 'center', backgroundColor: '#374777' }}>
```
Comment (Ignacio-Sormani) [inline client/src/components/layout/sidenav/Sidenav.jsx]: Write the styles in the css files and not as a prop if it's possible

File: client/src/components/pages/companies/Companies.jsx [hunk 3]
```diff
@@ -0,0 +1,45 @@
+import React, { useState } from 'react';
+import companies from './companies.json';
+import CompaniesTable from './companiesTable.jsx';
+
```
Comment (Ignacio-Sormani) [inline client/src/components/pages/companies/Companies.jsx]: Remove this empty line

File: client/src/components/pages/companies/Companies.jsx [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+import styles from './companies.module.css';
+import AddCompany from './addCompany';
+
```
Comment (Ignacio-Sormani) [inline client/src/components/pages/companies/Companies.jsx]: Only leave one empty line here

File: client/src/components/pages/companies/Companies.jsx [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+
+const Companies = () => {
+
```
Comment (Ignacio-Sormani) [inline client/src/components/pages/companies/Companies.jsx]: Remove this empty line

File: client/src/components/pages/companies/Companies.jsx [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+  // DELETE COMPANY
+  const deleteCompany = (id) => {
+    const companiesFilter = company.filter(company => company.id !== id);
+    setCompany(companiesFilter);
+    console.log('eliminado');
+  }
+  
+  return (
+    <div className={styles.container} >
+      {/* <h1>Companies Page</h1> */}
```
Comment (Ignacio-Sormani) [inline client/src/components/pages/companies/Companies.jsx]: Remove comments if you don't plan to uncomment it

File: client/src/components/pages/companies/addCompany.jsx [hunk 7]
```diff
@@ -0,0 +1,75 @@
+import React, { useState } from 'react';
+import Button from '@material-ui/core/Button';
+import AddBoxIcon from '@material-ui/icons/AddBox';
+import CheckCircleIcon from '@material-ui/icons/CheckCircle';
+import CancelIcon from '@material-ui/icons/Cancel';import TextField from '@material-ui/core/TextField';
```
Comment (Ignacio-Sormani) [inline client/src/components/pages/companies/addCompany.jsx]: Write each import in a new line

File: client/src/components/pages/companies/addCompany.jsx [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+import Modal from '@material-ui/core/Modal';
+import styles from './forms.module.css';
+
+const AddCompany = (props) => {
+
```
Comment (Ignacio-Sormani) [inline client/src/components/pages/companies/addCompany.jsx]: You can delete this empty line

File: client/src/components/pages/companies/addCompany.jsx [hunk 9]
```diff
@@ -0,0 +1,75 @@
...
+    e.preventDefault();
+    if (company.name && company.email && company.address && company.buildings  ) {
+       handleChange(e, props.addCompany(company));
+    }
+    setOpen(false);
+  };
+
+  const handleOpen = () => {
+    setOpen(true);
+    console.log('HANDLEOPEN');
```
Comment (Ignacio-Sormani) [inline client/src/components/pages/companies/addCompany.jsx]: Remove this console.log

File: client/src/components/pages/companies/addCompany.jsx [hunk 10]
```diff
@@ -0,0 +1,75 @@
...
+  };
+
+  const handleClose = () => {
+    setOpen(false);
+  };
+
+  return(
+    <div className={styles.container}>
+      <Button variant="contained"
```
Comment (Ignacio-Sormani) [inline client/src/components/pages/companies/addCompany.jsx]: Be careful with the indentation here. If the line is too long, each prop should be indented in a new line

File: client/src/components/pages/companies/addCompany.jsx [hunk 11]
```diff
@@ -0,0 +1,75 @@
...
+                onClick={handleOpen}
+                >
+                  <AddBoxIcon />
+      </Button>
+      <Modal 
+      open={open}
+      onClose={handleClose}
+      >
+        <form className={styles.formModal} >
+        <TextField id="outlined-basic" label="NAME" variant="outlined" onChange={handleChange} value={company.name} name="name" />
```
Comment (Ignacio-Sormani) [inline client/src/components/pages/companies/addCompany.jsx]: These lines are too long

File: client/src/components/pages/companies/companiesTable.jsx [hunk 12]
```diff
@@ -0,0 +1,60 @@
...
+import TableBody from '@material-ui/core/TableBody';
+import TableCell from '@material-ui/core/TableCell';
+import TableContainer from '@material-ui/core/TableContainer';
+import TableHead from '@material-ui/core/TableHead';
+import TableRow from '@material-ui/core/TableRow';
+import Paper from '@material-ui/core/Paper';
+import Button from '@material-ui/core/Button';
+import EditIcon from '@material-ui/icons/Edit';
+import DeleteIcon from '@material-ui/icons/Delete';
+
```
Comment (Ignacio-Sormani) [inline client/src/components/pages/companies/companiesTable.jsx]: Remove this empty line

File: client/src/components/pages/companies/companiesTable.jsx [hunk 13]
```diff
@@ -0,0 +1,60 @@
...
+import AddCompany from './addCompany';
+
```
Comment (Ignacio-Sormani) [inline client/src/components/pages/companies/companiesTable.jsx]: Remove this empty line

File: client/src/components/pages/companies/companiesTable.jsx [hunk 14]
```diff
@@ -0,0 +1,60 @@
...
+import styles from './companiesTable.module.css';
+
+const CompaniesTable = (props) => {
+
```
Comment (Ignacio-Sormani) [inline client/src/components/pages/companies/companiesTable.jsx]: You can remove this empty line

**Review by Ignacio-Sormani — APPROVED** [review]

**Review by giulice — APPROVED** [review]

**Pull request merged**